jetty-start/src/main/java/org/eclipse/jetty/start/JarLicense.java

```diff
@@ -0,0 +1,123 @@
+//
+// ========================================================================
+// Copyright (c) 1995 Mort Bay Consulting Pty Ltd and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// https://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+// ========================================================================
+//
+
+package org.eclipse.jetty.start;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.util.List;
+import java.util.Objects;
+import java.util.function.Function;
+import java.util.jar.JarFile;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.zip.ZipEntry;
+
+public class JarLicense
+{
+    private static final Pattern MANIFEST_MF_LICENSE = Pattern.compile("Bundle-License: https?://([^ ,]+)");
+
+    /**
+     * Get the license from the given jar file.
+     * Method tries to find the license from 3 places
+     * 1. MANIFEST.MF
+     * 2. pom.xml
+     * 3. LICENSE file
+     * @param file Jar File
+     * @return License name or (none specified) if license is absent
+     */
+    public static String getLicense(File file)
+    {
+        String nonSpecified = "(none specified)";
+        try (JarFile jar = new JarFile(file))
+        {
+            return findFirstNonNullLicenseOr(
+                List.of(
+                    JarLicense::findFromManifest,
+                    JarLicense::findFromPOM,
+                    JarLicense::findFromLicenseFile
```

Comment:
There should also be a `JarLicense::findFromSPDXIndicator`.
And it should report all of them, not just the first one it finds.

Now, you can present them in order though.
1. `SPDX-License-Identifier` results
2. POM
3. Manifest
4. License Files (NOTICE.*, LICENSE.*) - the output of these should indicate (via URI) where they came from.
If there is any order, it should be present